db/collection_index.go

```diff
@@ -588,3 +592,18 @@ func generateIndexName(col client.Collection, fields []client.IndexedFieldDescri
 	}
 	return sb.String()
 }
+
+func isUpdatingIndexedFields(index CollectionIndex, doc *client.Document) bool {
```

Comment:
suggestion: I think it's better to compare 2 docs here. What if an update is requested, but to the same value? Will `val.IsDirty` return true?

You can instead pass here 2 docs and do something like:
```go
for _, indexedFields := range index.Description().Fields {
	newVal := newDoc.GetValue(indexedFields.Name)
	oldVal := oldDoc.GetValue(indexedFields.Name)
	if newVal.Value() != oldVal.Value() {
		return true
	}
}
```
This way you don't need to iterate through all `doc.Fields()` but only indexed ones.

Reply:
> What if an update is requested, but to the same value? Will val.IsDirty return true?
Yes.

I like that suggestion. It will ensure we don't try to update the index if the indexed field values are set to the same value 👍

Reply:
✅

Reply:
I'll wait for you to check the change before merging.

Reply:
@islamaliev still waiting for you to check the change :)